SoftwareGuide/Latex/DevelopmentGuidelines/CreateAModule.tex

```diff
@@ -929,6 +930,151 @@ \subsubsection{Wrapping Tests}
+    under the ITK build tree. These C++ files explicitly implement the template
+    instantiations defined in the class \code{.wrap} files in the source tree.
+    Debug symbols will be generated for these files.
+  \item SWIG compiled code. For Python wrappings these are generated as Python
+    \code{.pyd} binaries and Python \code{.py} modules at
+    \code{Wrapping/Generators/Python/itk} in the ITK build tree.
+  \item Additional Python configuration files are generated in the
+    \code{Wrapping/Generators/Python} directory and its subdirectories.
+    \code{WrapITK.pth} provides the path for a Python environment to find the ITK
+    module, while \code{\_\_init\_\_.py} allows the module to be loaded correctly at
+    runtime. \code{<module\_name>Config.py} defines module dependencies and class
+    template definitions, while \code{<module\_name>\_snake\_case.py} maps
+    C++-style filter pipeline executions to Pythonic snake case functions.
+\end{itemize}
+
+ITK Python pre-compiled wheels may be obtained from the PyPi package index and contain only
+Python modules and SWIG compiled binaries.
+
+\begin{minted}[baselinestretch=1,fontsize=\footnotesize,linenos=false,bgcolor=ltgray]{bash}
+python -m pip install itk
+\end{minted}
+
+
+\subsubsection{Python Runtime Tracing}
+
+ITK Python contains glue behavior to make ITK classes behave in a Pythonic manner,
+such as querying object attributes, joining class names to template instantiations, and more.
+Python-specific behavior in ITK Python may be investigated with the
+Python Debugger module \code{pdb}.
+
+Tracing can be performed on a function call with \code{pdb.run}, or by editing ITK Python
+files to add a \code{pdb.set\_trace()} statement inline. In the following code snippet
+an image is loaded and the debugger is set up to trace through an image cast operation.
+
+\begin{minted}[baselinestretch=1,fontsize=\footnotesize,linenos=false,bgcolor=ltgray]{bash}
+(venv-itk) > python
+>>> import itk
+>>> import pdb
+>>> image = itk.imread(r'myimage.mha',pixel_type=itk.F)
+>>> pdb.runeval('itk.cast_image_filter(image,ttype=[type(image),itk.Image[itk.UC,2]])')
+> <string>(1)<module>()
+(Pdb)
+\end{minted}
+
+At this point the debugger can step into and through Python code for translating ITK
+type names and getting the correct template instantiations to run the cast operation. More
+information on Python debugger commands can be found at
+\href{https://docs.python.org/3/library/pdb.html}{https://docs.python.org/3/library/pdb.html}.
+
+While the Python debugger is useful, it does not allow us to examine implementation
+details of ITK classes on its own. A deeper investigation may be necessary for localizing
+errors in ITK classes.
+
+
+\subsubsection{SWIG C++ Runtime Tracing}
+
+As discussed in Section~\ref{subsubsec:SwigArchitecture}, SWIG wrapping generates
+C++ source files managing template implementations during ITK compilation.
```

Comment:
managing template implementations -> and manage the interface and ownership semantics between Python objects and C++ objects